pkg/placement/placement_client.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package placement
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/store/pdtypes"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/httputil"
+	"github.com/pingcap/tiflow/pkg/retry"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	updateTypeDel = iota
+	updateTypeAdd
+)
+
+const (
+	batchUpdatePrefix = "/pd/api/v1/config/rules/batch"
+	groupQueryPrefix  = "/pd/api/v1/config/rules/group"
+	idQueryPrefix     = "/pd/api/v1/config/rule"
+
+	addMetaJson = `[{
+		"group_id": "pd",
+		"id": "cdcmeta",
+		"index": 20000,
+		"override": true,
+		"start_key": "6d00000000000000f8",
+		"end_key": "6e00000000000000f8",
+		"role": "voter",
+		"count": 3,
+		"action": "add",
+		"delete_by_id_prefix": false
+	}]`
+
+	delCDCGroupJson = `[{
+		"group_id": "pd",
+		"id": "cdcmeta",
+		"action": "del",
+		"delete_by_id_prefix": true
+	}]`
+)
+
+var placmentRuleCheckInterval = 20 * time.Second
```

Comment:
Can it be an argument in `NewPlacementClient`? So we don't need the global setting.

Reply:
Good advice!